### ISSUE TITLE
DR1: Delta HF Integrale not recognized

### ISSUE THREAD

**Issue by fenugrec** [soong-construction/dirt-rally-time-recorder#50]
Hi, I just checkout most recent master (haven't played since ~ may), and trying the HF Integrale gives this in timerecord.log:

```
2020-12-28 21:39:01,061 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:39:01,083 INFO [timerecorder.statsProcessor] TRACK: Fourketa Kourva
2020-12-28 21:39:01,083 WARNING [timerecorder.databaseAccess] Failed to identify car
2020-12-28 21:39:01,083 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 183.2595672607422 - 785.398193359375
2020-12-28 21:39:01,083 DEBUG [timerecorder.statsProcessor] q1587780213.5.-1.started
2020-12-28 21:42:08,723 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:42:08,872 DEBUG [timerecorder.statsProcessor] q1587780213.5.-1.time:173.62|s
2020-12-28 21:42:08,872 DEBUG [timerecorder.statsProcessor] q1587780213.5.-1.topspeed:151.2|kph
2020-12-28 21:42:08,872 INFO [timerecorder.statsProcessor] Completed stage in 02:53.617.
```

But I checked the setup-dr1.sql that the values match except for Idle RPM: 
`INSERT INTO cars (id, name, maxrpm, idlerpm) VALUES (14, 'Delta HF Integrale', 785.398193359375, 183.25953674316406);`

Hmm is my db broken / needing migration ?

**Comment by maxbechtold** [comment]
Hey fenugrec, thanks for submitting. 

I seem to remember this happening once for me as well, I think it was DiRT Rally. It seems the idle RPM is off by what could be a rather large rounding error. 
Also, it's curious that this happened after a restart (where the game might not "reset" the car properly). I should look into this again, although I don't have much hope to find a solution if the data is taken correctly from the UDP telemetry. Would it be possible for you to include further lines from the log, e.g. from the complete session on this day?

The `ambiguousResultHandler` couldn't provide any update scripts because it's not really an ambiguity. 

Your `dirtrally-laptimes.db` should now contain an incomplete entry for this run, but this doesn't hurt and will be ignored e.g. when you export your times using the provided script. If you're familiar with the sqlite3 CLI, you can fix the database yourself with using the commands below (note the time is the finish time in seconds, cf. log output), best back it up beforehand:
```
.open dirtrally-laptimes.db
select * from laptimes where abs(time-173.62)<0.01; -- Check that this finds exactly the ambigous recording
update laptimes set car=14 where abs(time-173.62)<0.01;
.exit
```

Let me know if you need further help.

**Comment by maxbechtold** [comment]
Oh and although it cannot be the issue here, but in general it's safer to use the [latest release](https://github.com/soong-construction/dirt-rally-time-recorder/releases/latest), either the bundle or by cloning the repo from there.

**Comment by fenugrec** [comment]
Thanks!
Indeed the idle RPM is "close" - comparing float values for equality is always problematic; how is the raw value transmitted ?

> further lines from the log

sure thing, see below.
I'm too lazy to fix all the incomplete entries (they were lousy times anyway), would it be simpler to purge them or just leave them floating in there ?

```
2020-12-28 20:23:41,130 INFO [timerecorder.timerecord] Starting dirt-rally-time-recorder 2.10.0
2020-12-28 20:23:41,198 DEBUG [timerecorder.config] Loading config
2020-12-28 20:23:41,209 INFO [timerecorder.database] Found 84 tracks and 36 cars
2020-12-28 20:23:41,209 DEBUG [timerecorder.updateScriptHandler] Cleaning up update scripts...
2020-12-28 20:23:41,223 INFO [timerecorder.database] Checking laptimes database
2020-12-28 20:23:41,223 INFO [timerecorder.receiver] Waiting for data on 127.0.0.1:20777
2020-12-28 20:24:48,679 DEBUG [timerecorder.receiver] Receiving data on 127.0.0.1:20777
2020-12-28 20:24:48,680 INFO [timerecorder.statsProcessor] TRACK: Hamra
2020-12-28 20:24:48,680 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 20:24:48,680 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 20:24:48,680 DEBUG [timerecorder.statsProcessor] q1587780213.107.-1.started
2020-12-28 20:24:48,681 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:24:48,681 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:24:48,681 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 20:31:57,218 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 20:31:57,218 INFO [timerecorder.ambiguousResultHandler]  ==> 1609205517_Hamra_FordFocusRSRally2007.bat
2020-12-28 20:31:57,256 INFO [timerecorder.ambiguousResultHandler]  ==> 1609205517_Hamra_CitroenC4Rally2007.bat
2020-12-28 20:31:57,257 INFO [timerecorder.ambiguousResultHandler]  ==> 1609205517_Hamra_HyundaiRally.bat
2020-12-28 20:31:57,390 DEBUG [timerecorder.statsProcessor] q1587780213.107.-1.time:406.78|s
2020-12-28 20:31:57,390 DEBUG [timerecorder.statsProcessor] q1587780213.107.-1.topspeed:180.0|kph
2020-12-28 20:31:57,390 INFO [timerecorder.statsProcessor] Completed stage in 06:46.783.
2020-12-28 20:38:57,679 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 20:38:57,696 INFO [timerecorder.statsProcessor] TRACK: Älgsjön
2020-12-28 20:38:57,696 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 20:38:57,696 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 20:38:57,696 DEBUG [timerecorder.statsProcessor] q1587780213.101.-1.started
2020-12-28 20:38:57,696 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:38:57,696 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:38:57,697 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 20:42:04,278 INFO [timerecorder.timerecord] Starting dirt-rally-time-recorder 2.10.0
2020-12-28 20:42:04,396 DEBUG [timerecorder.config] Loading config
2020-12-28 20:42:04,396 INFO [timerecorder.database] Found 84 tracks and 36 cars
2020-12-28 20:42:04,397 DEBUG [timerecorder.updateScriptHandler] Cleaning up update scripts...
2020-12-28 20:42:04,397 INFO [timerecorder.database] Checking laptimes database
2020-12-28 20:42:04,398 INFO [timerecorder.receiver] Waiting for data on 127.0.0.1:20777
2020-12-28 20:42:09,561 DEBUG [timerecorder.receiver] Receiving data on 127.0.0.1:20777
2020-12-28 20:42:09,562 INFO [timerecorder.statsProcessor] TRACK: Kailajärvi
2020-12-28 20:42:09,562 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 20:42:09,563 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 20:42:09,563 DEBUG [timerecorder.statsProcessor] q1587780213.81.-1.started
2020-12-28 20:42:09,563 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:42:09,563 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:42:09,563 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 20:43:49,610 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 20:43:49,630 INFO [timerecorder.statsProcessor] TRACK: Kailajärvi
2020-12-28 20:43:49,630 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 20:43:49,630 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 20:43:49,630 DEBUG [timerecorder.statsProcessor] q1587780213.81.-1.started
2020-12-28 20:43:49,631 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:43:49,631 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:43:49,631 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 20:47:37,840 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 20:47:37,841 INFO [timerecorder.ambiguousResultHandler]  ==> 1609206457_Kailajärvi_FordFocusRSRally2007.bat
2020-12-28 20:47:37,841 INFO [timerecorder.ambiguousResultHandler]  ==> 1609206457_Kailajärvi_CitroenC4Rally2007.bat
2020-12-28 20:47:37,841 INFO [timerecorder.ambiguousResultHandler]  ==> 1609206457_Kailajärvi_HyundaiRally.bat
2020-12-28 20:47:37,981 DEBUG [timerecorder.statsProcessor] q1587780213.81.-1.time:216.82|s
2020-12-28 20:47:37,982 DEBUG [timerecorder.statsProcessor] q1587780213.81.-1.topspeed:190.8|kph
2020-12-28 20:47:37,982 INFO [timerecorder.statsProcessor] Completed stage in 03:36.824.
2020-12-28 20:48:14,085 DEBUG [timerecorder.respawnTracker] RECOVER
2020-12-28 20:48:14,103 INFO [timerecorder.statsProcessor] TRACK: Naarajärvi
2020-12-28 20:48:14,110 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 20:48:14,110 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 20:48:14,110 DEBUG [timerecorder.statsProcessor] q1587780213.83.-1.started
2020-12-28 20:48:14,110 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:48:14,110 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:48:14,110 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 20:51:49,950 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 20:51:49,950 INFO [timerecorder.ambiguousResultHandler]  ==> 1609206709_Naarajärvi_FordFocusRSRally2007.bat
2020-12-28 20:51:49,950 INFO [timerecorder.ambiguousResultHandler]  ==> 1609206709_Naarajärvi_CitroenC4Rally2007.bat
2020-12-28 20:51:49,951 INFO [timerecorder.ambiguousResultHandler]  ==> 1609206709_Naarajärvi_HyundaiRally.bat
2020-12-28 20:51:50,073 DEBUG [timerecorder.statsProcessor] q1587780213.83.-1.time:221.64|s
2020-12-28 20:51:50,074 DEBUG [timerecorder.statsProcessor] q1587780213.83.-1.topspeed:180.0|kph
2020-12-28 20:51:50,074 INFO [timerecorder.statsProcessor] Completed stage in 03:41.641.
2020-12-28 20:52:36,355 INFO [timerecorder.statsProcessor] TRACK: Hämelahti
2020-12-28 20:52:36,355 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 20:52:36,355 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 20:52:36,355 DEBUG [timerecorder.statsProcessor] q1587780213.92.-1.started
2020-12-28 20:52:36,355 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:52:36,355 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 20:52:36,355 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 20:59:36,601 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 20:59:36,602 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207176_Hämelahti_FordFocusRSRally2007.bat
2020-12-28 20:59:36,602 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207176_Hämelahti_CitroenC4Rally2007.bat
2020-12-28 20:59:36,602 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207176_Hämelahti_HyundaiRally.bat
2020-12-28 20:59:36,779 DEBUG [timerecorder.statsProcessor] q1587780213.92.-1.time:402.80|s
2020-12-28 20:59:36,780 DEBUG [timerecorder.statsProcessor] q1587780213.92.-1.topspeed:187.2|kph
2020-12-28 20:59:36,780 INFO [timerecorder.statsProcessor] Completed stage in 06:42.800.
2020-12-28 21:00:20,491 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:00:20,501 INFO [timerecorder.statsProcessor] TRACK: Kontinjärvi
2020-12-28 21:00:20,502 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 21:00:20,502 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 21:00:20,502 DEBUG [timerecorder.statsProcessor] q1587780213.91.-1.started
2020-12-28 21:00:20,502 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 21:00:20,502 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 21:00:20,502 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 21:07:32,808 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:07:32,808 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207652_Kontinjärvi_FordFocusRSRally2007.bat
2020-12-28 21:07:32,808 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207652_Kontinjärvi_CitroenC4Rally2007.bat
2020-12-28 21:07:32,808 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207652_Kontinjärvi_HyundaiRally.bat
2020-12-28 21:07:32,955 DEBUG [timerecorder.statsProcessor] q1587780213.91.-1.time:424.10|s
2020-12-28 21:07:32,955 DEBUG [timerecorder.statsProcessor] q1587780213.91.-1.topspeed:180.0|kph
2020-12-28 21:07:32,955 INFO [timerecorder.statsProcessor] Completed stage in 07:04.099.
2020-12-28 21:08:12,612 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:08:12,640 INFO [timerecorder.statsProcessor] TRACK: Paskuri
2020-12-28 21:08:12,641 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 21:08:12,641 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 21:08:12,641 DEBUG [timerecorder.statsProcessor] q1587780213.82.-1.started
2020-12-28 21:08:12,641 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 21:08:12,641 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 21:08:12,641 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 21:12:20,427 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:12:20,427 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207940_Paskuri_FordFocusRSRally2007.bat
2020-12-28 21:12:20,427 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207940_Paskuri_CitroenC4Rally2007.bat
2020-12-28 21:12:20,427 INFO [timerecorder.ambiguousResultHandler]  ==> 1609207940_Paskuri_HyundaiRally.bat
2020-12-28 21:12:20,567 DEBUG [timerecorder.statsProcessor] q1587780213.82.-1.time:202.56|s
2020-12-28 21:12:20,567 DEBUG [timerecorder.statsProcessor] q1587780213.82.-1.topspeed:190.8|kph
2020-12-28 21:12:20,568 INFO [timerecorder.statsProcessor] Completed stage in 03:22.558.
2020-12-28 21:12:55,511 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:12:55,521 INFO [timerecorder.statsProcessor] TRACK: Jyrkysjärvi
2020-12-28 21:12:55,522 WARNING [timerecorder.databaseAccess] Ambiguous car data, 3 matches
2020-12-28 21:12:55,522 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 209.43939208984375 - 733.038330078125
2020-12-28 21:12:55,522 DEBUG [timerecorder.statsProcessor] q1587780213.84.-1.started
2020-12-28 21:12:55,522 INFO [timerecorder.statsProcessor] Ford Focus RS Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 21:12:55,522 INFO [timerecorder.statsProcessor] Citroen C4 Rally 2007: PADDLE shifting, 6 speed, with HANDBRAKE
2020-12-28 21:12:55,522 INFO [timerecorder.statsProcessor] Hyundai Rally: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2020-12-28 21:16:22,970 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:16:22,970 INFO [timerecorder.ambiguousResultHandler]  ==> 1609208182_Jyrkysjärvi_FordFocusRSRally2007.bat
2020-12-28 21:16:22,971 INFO [timerecorder.ambiguousResultHandler]  ==> 1609208182_Jyrkysjärvi_CitroenC4Rally2007.bat
2020-12-28 21:16:22,971 INFO [timerecorder.ambiguousResultHandler]  ==> 1609208182_Jyrkysjärvi_HyundaiRally.bat
2020-12-28 21:16:23,098 DEBUG [timerecorder.statsProcessor] q1587780213.84.-1.time:198.93|s
2020-12-28 21:16:23,098 DEBUG [timerecorder.statsProcessor] q1587780213.84.-1.topspeed:180.0|kph
2020-12-28 21:16:23,099 INFO [timerecorder.statsProcessor] Completed stage in 03:18.925.
2020-12-28 21:17:21,401 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:17:21,414 INFO [timerecorder.statsProcessor] TRACK: Bidno Moorland Reverse
2020-12-28 21:17:21,414 WARNING [timerecorder.databaseAccess] Ambiguous car data, 2 matches
2020-12-28 21:17:21,414 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 125.66371154785156 - 890.117919921875
2020-12-28 21:17:21,414 DEBUG [timerecorder.statsProcessor] q1587780213.63.-1.started
2020-12-28 21:17:21,414 INFO [timerecorder.statsProcessor] Ford RS200: H-PATTERN shifting, 5 speed, with manual CLUTCH
2020-12-28 21:17:21,414 INFO [timerecorder.statsProcessor] Lancia 037 Evo 2: H-PATTERN shifting, 5 speed, with manual CLUTCH, with HANDBRAKE
2020-12-28 21:18:56,417 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:18:56,430 INFO [timerecorder.statsProcessor] TRACK: Bidno Moorland Reverse
2020-12-28 21:18:56,430 WARNING [timerecorder.databaseAccess] Ambiguous car data, 2 matches
2020-12-28 21:18:56,430 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 125.66371154785156 - 890.117919921875
2020-12-28 21:18:56,431 DEBUG [timerecorder.statsProcessor] q1587780213.63.-1.started
2020-12-28 21:18:56,431 INFO [timerecorder.statsProcessor] Ford RS200: H-PATTERN shifting, 5 speed, with manual CLUTCH
2020-12-28 21:18:56,431 INFO [timerecorder.statsProcessor] Lancia 037 Evo 2: H-PATTERN shifting, 5 speed, with manual CLUTCH, with HANDBRAKE
2020-12-28 21:20:17,922 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:20:17,938 INFO [timerecorder.statsProcessor] TRACK: Bidno Moorland Reverse
2020-12-28 21:20:17,938 WARNING [timerecorder.databaseAccess] Ambiguous car data, 2 matches
2020-12-28 21:20:17,938 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 125.66371154785156 - 890.117919921875
2020-12-28 21:20:17,938 DEBUG [timerecorder.statsProcessor] q1587780213.63.-1.started
2020-12-28 21:20:17,938 INFO [timerecorder.statsProcessor] Ford RS200: H-PATTERN shifting, 5 speed, with manual CLUTCH
2020-12-28 21:20:17,938 INFO [timerecorder.statsProcessor] Lancia 037 Evo 2: H-PATTERN shifting, 5 speed, with manual CLUTCH, with HANDBRAKE
2020-12-28 21:21:33,089 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:21:33,101 INFO [timerecorder.statsProcessor] TRACK: Bidno Moorland Reverse
2020-12-28 21:21:33,102 WARNING [timerecorder.databaseAccess] Ambiguous car data, 2 matches
2020-12-28 21:21:33,102 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 125.66371154785156 - 890.117919921875
2020-12-28 21:21:33,102 DEBUG [timerecorder.statsProcessor] q1587780213.63.-1.started
2020-12-28 21:21:33,102 INFO [timerecorder.statsProcessor] Ford RS200: H-PATTERN shifting, 5 speed, with manual CLUTCH
2020-12-28 21:21:33,102 INFO [timerecorder.statsProcessor] Lancia 037 Evo 2: H-PATTERN shifting, 5 speed, with manual CLUTCH, with HANDBRAKE
2020-12-28 21:25:11,161 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:25:11,161 INFO [timerecorder.ambiguousResultHandler]  ==> 1609208711_BidnoMoorlandReverse_FordRS200.bat
2020-12-28 21:25:11,161 INFO [timerecorder.ambiguousResultHandler]  ==> 1609208711_BidnoMoorlandReverse_Lancia037Evo2.bat
2020-12-28 21:25:11,276 DEBUG [timerecorder.statsProcessor] q1587780213.63.-1.time:207.20|s
2020-12-28 21:25:11,277 DEBUG [timerecorder.statsProcessor] q1587780213.63.-1.topspeed:151.2|kph
2020-12-28 21:25:11,277 INFO [timerecorder.statsProcessor] Completed stage in 03:27.195.
2020-12-28 21:26:13,820 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:26:13,881 INFO [timerecorder.statsProcessor] TRACK: Anodou Farmakas
2020-12-28 21:26:13,881 WARNING [timerecorder.databaseAccess] Failed to identify car
2020-12-28 21:26:13,881 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 183.2595672607422 - 785.398193359375
2020-12-28 21:26:13,881 DEBUG [timerecorder.statsProcessor] q1587780213.12.-1.started
2020-12-28 21:27:33,813 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:27:33,826 INFO [timerecorder.statsProcessor] TRACK: Anodou Farmakas
2020-12-28 21:27:33,827 WARNING [timerecorder.databaseAccess] Failed to identify car
2020-12-28 21:27:33,827 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 183.2595672607422 - 785.398193359375
2020-12-28 21:27:33,827 DEBUG [timerecorder.statsProcessor] q1587780213.12.-1.started
2020-12-28 21:34:14,008 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:34:14,170 DEBUG [timerecorder.statsProcessor] q1587780213.12.-1.time:381.16|s
2020-12-28 21:34:14,171 DEBUG [timerecorder.statsProcessor] q1587780213.12.-1.topspeed:162.0|kph
2020-12-28 21:34:14,171 INFO [timerecorder.statsProcessor] Completed stage in 06:21.158.
2020-12-28 21:35:04,577 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:35:04,591 INFO [timerecorder.statsProcessor] TRACK: Ampelonas Ormi
2020-12-28 21:35:04,591 WARNING [timerecorder.databaseAccess] Failed to identify car
2020-12-28 21:35:04,591 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 183.2595672607422 - 785.398193359375
2020-12-28 21:35:04,591 DEBUG [timerecorder.statsProcessor] q1587780213.1.-1.started
2020-12-28 21:38:25,392 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:38:25,513 DEBUG [timerecorder.statsProcessor] q1587780213.1.-1.time:192.41|s
2020-12-28 21:38:25,513 DEBUG [timerecorder.statsProcessor] q1587780213.1.-1.topspeed:154.8|kph
2020-12-28 21:38:25,513 INFO [timerecorder.statsProcessor] Completed stage in 03:12.412.
2020-12-28 21:39:01,061 DEBUG [timerecorder.respawnTracker] RESTART
2020-12-28 21:39:01,083 INFO [timerecorder.statsProcessor] TRACK: Fourketa Kourva
2020-12-28 21:39:01,083 WARNING [timerecorder.databaseAccess] Failed to identify car
2020-12-28 21:39:01,083 DEBUG [timerecorder.databaseAccess] Idle/Max RPM: 183.2595672607422 - 785.398193359375
2020-12-28 21:39:01,083 DEBUG [timerecorder.statsProcessor] q1587780213.5.-1.started
2020-12-28 21:42:08,723 INFO [timerecorder.ambiguousResultHandler] Please run one of the scripts below to link the recorded laptime to the correct car:
2020-12-28 21:42:08,872 DEBUG [timerecorder.statsProcessor] q1587780213.5.-1.time:173.62|s
2020-12-28 21:42:08,872 DEBUG [timerecorder.statsProcessor] q1587780213.5.-1.topspeed:151.2|kph
2020-12-28 21:42:08,872 INFO [timerecorder.statsProcessor] Completed stage in 02:53.617.
```

**Comment by fenugrec** [comment]
(just noting this as a reminder to me or others in this situation) 
-- since I only had one unrecognized car, I tweaked the commands you suggested to 
```
.open dirtrally-laptimes.db
select * from laptimes where car == -1;
update laptimes set car=14 where car == -1;
.exit
```
To see the car # definitions I double-checked with, 
```
.open dirtrally-lb.db
.dump cars
```

**Comment by maxbechtold** [comment]
> Indeed the idle RPM is "close" - comparing float values for equality is always problematic; how is the raw value transmitted ?

Hence the comparison is pretty lenient, cf. database.py:91 (allows up to 1.0 delta). 
Also, it checks for the top gear (which helps in telling apart cars with identical RPM) - which is configured wrongly in the setup data (5 instead of 6 gears in-game). So it cannot identify the Delta HF Integrale in any case. I will fix this in version 1.10.1.


> I'm too lazy to fix all the incomplete entries (they were lousy times anyway), would it be simpler to purge them or just leave them floating in there ?

If you still have the update scripts for the previous stages and remember the actual car, use them. Otherwise I recommend ignoring the incomplete entries, as they are filtered out by the bat scripts.

Looking at the log I was wrong, the RESTART marker is also issued at the regular start of a stage. 

> -- since I only had one unrecognized car, I tweaked the commands you suggested to

Sure, you seem to be able to handle SQL. Still, I believe the timerecorder could log the incomplete update statement (the one used in the update scripts that are missing in the no-car-found case), then it would be easier to consolidate the laptimes database. I'll track this in a separate issue.

**Comment by fenugrec** [comment]
`(5 instead of 6 gears in-game). So it cannot identify the Delta HF Integrale in any case. I will fix this in version 1.10.1.`

Ah good catch !
Since that was the root issue here and fixed in master , this can be closed as far as I'm concerned.

**Comment by maxbechtold** [comment]
Fixed with https://github.com/soong-construction/dirt-rally-time-recorder/releases/tag/v2.10.1
